.github/workflows/deploy-book.yaml

```diff
@@ -0,0 +1,59 @@
+name: deploy-book
+
+# Run this when the master or main branch changes
+on:
+  push:
+    branches:
+    - 1-sdk-creation
+
+# This job installs dependencies, builds the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    permissions:
+      pages: write
+      id-token: write
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v2
+      with:
+        # Install a specific version of uv.
+        version: "0.4.10"
+        enable-cache: true
+        cache-dependency-glob: "uv.lock"
+
+    - name: "Set up Python"
+      uses: actions/setup-python@v5
+      with:
+        python-version-file: ".python-version"
+
+    - name: Install the project
+      run: uv sync --all-extras --dev
+
+    # (optional) Cache your executed notebooks between runs
+    # if you have config:
+    # execute:
+    #   execute_notebooks: cache
+    - name: cache executed notebooks
+      uses: actions/cache@v3
+      with:
+        path: _build/.jupyter_cache
+        key: jupyter-book-cache-${{ hashFiles('uv.lock') }}
+
+    # Build the book
+    - name: Build the book
+      run: |
+        uv run jupyter-book build docs/
+
+    # Upload the book's HTML as an artifact
+    - name: Upload artifact
```

Comment:
Why are you uploading as an artifact? I don't see you using the artifact in a different workflow, so it is redundant?